modules/launcher/src/test/java/org/mule/module/launcher/domain/DefaultMuleDomainTestCase.java

```diff
@@ -0,0 +1,183 @@
+import org.mule.api.transport.Connector;
+import org.mule.module.launcher.artifact.ArtifactClassLoader;
+import org.mule.module.launcher.descriptor.DomainDescriptor;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mule.module.launcher.domain.Domain.DOMAIN_CONFIG_FILE_LOCATION;
+
+public class DefaultMuleDomainTestCase
+{
+    private static final MuleContextFactory muleContextFactory = mock(MuleContextFactory.class);
+    private DomainDescriptor domainDescriptor;
+    private DomainClassLoaderRepository domainClassLoaderRepository;
+    private DefaultMuleDomain defaultMuleDomain;
+    private MuleContext muleContext;
+    private MuleRegistry muleRegistry;
+
+    private URL resource;
+    private ArtifactClassLoader artifactClassLoader;
+
+    @Before
+    public void setUp() throws Exception
+    {
+        domainDescriptor = mock(DomainDescriptor.class);
+        domainClassLoaderRepository = mock(DomainClassLoaderRepository.class);
+        artifactClassLoader = mock(ArtifactClassLoader.class);
+        when(domainClassLoaderRepository.getDomainClassLoader(domainDescriptor)).thenReturn(artifactClassLoader);
+        when(artifactClassLoader.findLocalResource(anyString())).thenReturn(null);
+        resource = getClass().getClassLoader().getResource("empty-domain-config.xml").toURI().toURL();
+        when(domainClassLoaderRepository.getDomainClassLoader(domainDescriptor)).thenReturn(artifactClassLoader);
+        when(artifactClassLoader.findLocalResource(DOMAIN_CONFIG_FILE_LOCATION)).thenReturn(resource);
+
+        muleContext = mock(MuleContext.class);
+        when(muleContextFactory.createMuleContext(any(List.class), any(MuleContextBuilder.class))).thenReturn(muleContext);
+
+        muleRegistry = mock(MuleRegistry.class);
+        when(muleContext.getRegistry()).thenReturn(muleRegistry);
+
+        defaultMuleDomain = new TestMuleDomain(domainClassLoaderRepository, domainDescriptor);
+        defaultMuleDomain.init();
+    }
+
+    @Test
+    public void testCancelStartOneConectorWithRetryPolicy()
+    {
+        // Given a default mule applicacion with 1 connector that has retryPolicy
+        List<Connector> connectors = new ArrayList<>();
+
+        RetryPolicyTemplate retryPolicyTemplate = addMockConnectorWithMockRetryPolicyToList(connectors);
+        when(muleRegistry.lookupObjects(Connector.class)).thenReturn(connectors);
+
+        // When cancelling start
+        defaultMuleDomain.cancelStart();
+
+        // Connectors's retry policy start is also cancelled
+        verify(retryPolicyTemplate).stopRetrying();
+    }
+
+    @Test
+    public void testCancelStartFiveConectorsWithRetryPolicies()
```

Comment:
Better to avoid magic numbers and stick to 2 if no extra logic (or cases) involved

Reply:
Applied the suggested changes